csharp/src/Google.Protobuf/Reflection/FieldDescriptor.cs

```diff
@@ -70,6 +70,21 @@ public sealed class FieldDescriptor : DescriptorBase, IComparable<FieldDescripto
         /// </summary>
         public string JsonName { get; }
 
+        /// <summary>
+        /// Indicates whether this field supports presence, either implicitly (e.g. due to it being a message
+        /// type field) or explicitly via Has/Clear members. If this returns true, it is safe to call
+        /// <see cref="IFieldAccessor.Clear(IMessage)"/> and <see cref="IFieldAccessor.HasValue(IMessage)"/>
+        /// on this field's accessor with a suitable message.
+        /// </summary>
+        public bool HasPresence =>
+            Extension != null ? !Extension.IsRepeated
```

Comment:
Isn't `Extension.IsRepeated` the same as `FieldDescriptor.IsRepeated`?

Reply:
Possibly. I didn't find it simple enough to reason about to convince myself of that - I figured this way may be a little redundant, but at least it's definitely right.